include/onnxruntime/core/graph/schema_registry.h

```diff
@@ -136,8 +138,30 @@ class SchemaRegistryManager : public onnxruntime::IOnnxRuntimeOpSchemaCollection
   @remarks The schema registry priority is the reverse of registration order. i.e. the last registry added will be
   searched first for a matching OpSchema.
   */
+  SchemaRegistryManager() {}
+
+  SchemaRegistryManager(const std::list<std::shared_ptr<IOnnxRuntimeOpSchemaCollection>>& registry_list) {
+    for (auto registry : registry_list) {
+      RegisterRegistry(registry);
+    }
+  }
+
   void RegisterRegistry(std::shared_ptr<IOnnxRuntimeOpSchemaCollection> registry);
 
+  void SetNewRegistryFunc(std::function<std::shared_ptr<IOnnxRuntimeOpSchemaCollection>(void*)> new_registry_func) {
+      new_registry_func_ = new_registry_func;
+  }
+
+  bool NewRegistry(void* info) {
```

Comment:
"new_registry_func_" will deal with the details, namely, it will create a JavaOp if found node a JavaOp.
The returned value will always be a new registry or nullptr.